Client/Actions/ViewVisitsAction.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using Client.HttpClients;
+using Client.IO.Abstract;
+using Client.Exceptions;
+using Common;
+
+namespace Client.Actions
+{
+    public class ViewVisitsAction : ActionBase
+    {
+        public ViewVisitsAction(MediClient client, IStreamIO streamIO) : base(client, streamIO)
+        {
+
+        }
+
+        public async override Task<ActionBase> Run()
+        {
+            List<Visit> visits = await GetVisits();
+            if (visits != null)
+            {
+                if (_client.User is Doctor)
+                {
+                    await PrintDoctorVisits(visits);
+                }
+                else
+                {
+                    await PrintPatientVisits(visits);
+                }
+            }
+
+            return new MainMenuAction(_client, _streamIO);
+        }
+
+        private async Task PrintDoctorVisits(List<Visit> visits)
+        {
```

Comment:
Extract print logic to other classes and inject a mapping between type and print method